File: PTA 1002 写出这个数.cpp
/*  PTA 1002 写出这个数
 *  2022/07/17
 * */
#include<stdio.h>
#include<math.h>
int main()
{
	char a[100];
	int sum = 0, c = 0, ss = 0;//c用于记录数字
	int gg = 0, z = 0;
	int mm = 0;
	gets(a);//输入到字符数组，算出总和
	for (int x = 0; a[x] != '\0'; x++)
	{
		sum = a[x] - 48 + sum;
	}
	ss = sum;
	//判断多少位
	while (sum != 0)
	{
		sum /= 10;
		c++;
	}
	// 开始输出
	for (int i = c; i > 0; i--)
	{
		z = ss / pow(10, i - 1);
		z = z % 10;
		if (i == 1)
		{
			switch (z)
			{
				case 0:printf("ling");
					break;
				case 1: printf("yi");
					break;
				case 2: printf("er");
					break;
				case 3: printf("san");
					break;
				case 4: printf("si");
					break;
				case 5: printf("wu");
					break;
				case 6: printf("liu");
					break;
				case 7: printf("qi");
					break;
				case 8: printf("ba");
					break;
				case 9: printf("jiu");
					break;
				
			}
		}
		else
		{
			switch (z)
			{
				case 0:printf("ling ");
					break;
				case 1: printf("yi ");
					break;
				case 2: printf("er ");
					break;
				case 3: printf("san ");
					break;
				case 4: printf("si ");
					break;
				case 5: printf("wu ");
					break;
				case 6: printf("liu ");
					break;
				case 7: printf("qi ");
					break;
				case 8: printf("ba ");
					break;
				case 9: printf("jiu ");
					break;
			}
		}
		
	}
	getchar();
	return 0;
}

File: PTA 1002 写出这个数.java
/// PTA 1002 写出这个数
/// 2023/01/05

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        int sum = 0;
        Scanner scanner = new Scanner(System.in);
        String str = scanner.nextLine();

        for (int i = 0; i < str.length(); i++)
            sum += str.charAt(i) - '0';

        str = String.valueOf(sum);
        for (int i = 0; i < str.length(); i++) {
            switch (str.charAt(i)) {
                case '0':
                    System.out.print("ling");
                    break;
                case '1':
                    System.out.print("yi");
                    break;
                case '2':
                    System.out.print("er");
                    break;
                case '3':
                    System.out.print("san");
                    break;
                case '4':
                    System.out.print("si");
                    break;
                case '5':
                    System.out.print("wu");
                    break;
                case '6':
                    System.out.print("liu");
                    break;
                case '7':
                    System.out.print("qi");
                    break;
                case '8':
                    System.out.print("ba");
                    break;
                case '9':
                    System.out.print("jiu");
                    break;
                default:
                    continue;
            }
            if (i < str.length() - 1)
                System.out.print(" ");
        }

    }
}
